src/ucs/type/spinlock.h

```diff
@@ -24,9 +24,24 @@ typedef struct ucs_spinlock {
 } ucs_spinlock_t;
 
 
-ucs_status_t ucs_spinlock_init(ucs_spinlock_t *lock);
+static inline ucs_status_t ucs_spinlock_init(ucs_spinlock_t *lock)
+{
+    int ret;
+
+    ret = pthread_spin_init(&lock->lock, 0);
+    if (ret != 0) {
+        return UCS_ERR_IO_ERROR;
+    }
 
-void ucs_spinlock_destroy(ucs_spinlock_t *lock);
+    lock->count = 0;
+    lock->owner = 0xfffffffful;
+    return UCS_OK;
+}
+
+static inline void ucs_spinlock_destroy(ucs_spinlock_t *lock)
+{
+    pthread_spin_destroy(&lock->lock);
+}
```

Comment:
Before the modification code is below. I have no idea how to check without `ucs_warn`
`ucs_assert` depends on UCS library, doesn't it?

```C
void ucs_spinlock_destroy(ucs_spinlock_t *lock)
{
    int ret;

    if (lock->count != 0) {
        ucs_warn("destroying spinlock %p with use count %d (owner: 0x%lx)",
                 lock, lock->count, lock->owner);
    }

    ret = pthread_spin_destroy(&lock->lock);
    if (ret != 0) {
        ucs_warn("failed to destroy spinlock %p: %s", lock, strerror(ret));
    }
}
```

Reply:
we can return ucs_status_t from ucs_spinlock_destroy()
the user would be responsible to check it

Reply:
Fixed. returns `ucs_status_t`;